bin/main.rb

```diff
@@ -1,36 +1,102 @@
-#!/usr/bin/env ruby
-# what do you need from player?
+require_relative '../lib/board.rb'
+require_relative '../lib/player.rb'
 require 'colorize'
-puts 'Welcome to Tic Tac Toe game'.blue
-puts 'Please enter name for player X'.red
-player_one = gets.chomp
-puts 'Please enter name for player O'.green
-player_two = gets.chomp
-board = %w[1 2 3 4 5 6 7 8 9]
-def grid(board)
-  puts "#{board[0]} | #{board[1]} | #{board[2]}"
-  puts '---------'
-  puts "#{board[3]} | #{board[4]} | #{board[5]}"
-  puts '---------'
-  puts "#{board[6]} | #{board[7]} | #{board[8]}"
+require 'artii'
+
+class Welcome < Player
+  arter = Artii::Base.new
+  puts arter.asciify('TIC TAC TOE')
+  puts 'Please enter name for player X'.cyan
+  puts ' '
+  @player_one = Player.new(gets.chomp)
+  puts ' '
+  puts 'Please enter name for player O'.green
+  puts ' '
+  @player_two = Player.new(gets.chomp)
+  puts ' '
+  puts 'PLAYERS'.red
+  puts ' '
+  puts 'Welcome '.white + @player_one.name.cyan + ' you are'.white + ' Player X'.cyan
+  puts ' '
+  puts 'Welcome '.white + @player_two.name.green + ' you are'.white + ' Player O'.green
+  puts ' '
+  puts 'PLAY TIME'.red
 end
-# Play time
-puts player_one.to_s.red + ' make your move'.white
-puts grid(board)
-# if move is valid return marked board and change player else return invalid move
-puts 'Invalid move'
-# check if win or tie with a loop
-while !win || !tie
+
+class Controller < Board
+  private
+
+  def validate(player_move)
+    @board.detect { |x| x == player_move.to_s }
+  end
+
+  def try_again
+    puts 'Enter a valid move'
+    valid_move
+  end
+
+  def valid_move
+    validate(gets.chomp) || try_again
+  end
+
+  def update_board
+    if @counter.even?
+      puts ' '
+      puts 'Player X '.cyan + 'Choose a number from 1 to 9'.white
+      puts ' '
+      @board[valid_move.to_i - 1] = 'X'.cyan
+    elsif @counter.odd?
+      puts 'Player O '.green + 'Choose a number from 1 to 9'.white
+      puts ' '
+      @board[valid_move.to_i - 1] = 'O'.green
+    end
+    @counter += 1
+    puts grid
+  end
+  WIN_COMBINATIONS = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [2, 4, 6],
+    [0, 4, 8]
+  ].freeze
+
+  def win(item)
+    result = false
+    WIN_COMBINATIONS.each do |w|
+      if item[w[0]] == 'X'.cyan && item[w[1]] == 'X'.cyan && item[w[2]] == 'X'.cyan
+        result = true
+      elsif item[w[0]] == 'O'.green && item[w[1]] == 'O'.green && item[w[2]] == 'O'.green
+        result = true
+      end
+    end
+    result
+  end
```

Comment:
- [ ] Try to use the DRY(don't repeat yourself) rule to check for the winning combinations rather than checking for the winning position for X and O separately.